src/cargo/core/compiler/build_runner/mod.rs

```diff
@@ -126,6 +126,23 @@ impl<'a, 'gctx> BuildRunner<'a, 'gctx> {
         })
     }
 
+    pub fn dry_run(mut self) -> CargoResult<Compilation<'gctx>> {
```

Comment:
Should we put a comment here about the expectation of how dryrun works? Supposedly we want to get the information of where dylibs and executables reside. So probably something like:

```suggestion
    /// Dry-run the compilation without actually running it.
    ///
    /// This is expected to collect information like the location of output artifacts.
    /// Please keep in sync with non-compilation part in [`BuildRunner::compile`].
    pub fn dry_run(mut self) -> CargoResult<Compilation<'gctx>> {
```

Reply:
I applied your suggestion